src/Count.php

```diff
@@ -0,0 +1,24 @@
+<?php
+
+namespace MaxGoryunov\SavingIterator\Src;
+
+/**
+ * Counts how many times something happened.
+ */
+interface Count
+{
+
+    /**
+     * Returns a counter with an incremented value.
+     *
+     * @return self
+     */
+    public function increment(): self;
```

Comment:
Fluent interface. Even though it does not require the implementing classes to be immutable this is assumed.